### PR TITLE
Pubsub spring integration test

### PR DESCRIPTION
**Pull request by creativeappdevamit**
# What and why?

# How to test?

# Trello
https://trello.com/c/126JzDkR/871-implement-spring-integration-for-pubsub-between-case-service-and-action-service

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/response/action/message/CaseNotificationReceiver.java [hunk 1]
```diff
@@ -3,66 +3,41 @@
...
+    try {
+      log.info("Mapping payload to CaseNotification object");
+      CaseNotification caseNotificationReceiver =
+          objectMapper.readValue(payload, CaseNotification.class);
+      log.info("Mapping successful, accepting action case notification");
+      caseNotificationService.acceptNotification(caseNotificationReceiver);
+    } catch (final IOException | CTPException e) {
+      log.with(e)
+          .error(
+              "Something went wrong while processing message received from PubSub for action case notification");
```
Comment (warrenbailey) [inline src/main/java/uk/gov/ons/ctp/response/action/message/CaseNotificationReceiver.java:40]: will this stop it nacking the message? 

I think spring relies on exceptions to know whether to nack or ack the messages

Comment (warrenbailey) [reply src/main/java/uk/gov/ons/ctp/response/action/message/CaseNotificationReceiver.java:40]: From here: https://cloud.spring.io/spring-cloud-gcp/multi/multi__spring_integration.html
```text
Automatic acking (AckMode.AUTO)

A message is acked with GCP Pub/Sub if the adapter sent it to the channel and no exceptions were thrown. If a RuntimeException is thrown while the message is processed, then the message is nacked.
```

Comment (creativeappdevamit) [reply src/main/java/uk/gov/ons/ctp/response/action/message/CaseNotificationReceiver.java:44]: you are right! changed it to Manual mode.

Comment (creativeappdevamit) [reply src/main/java/uk/gov/ons/ctp/response/action/message/CaseNotificationReceiver.java:44]: updated

**Review by insacuri — APPROVED** [review]

**Pull request merged**